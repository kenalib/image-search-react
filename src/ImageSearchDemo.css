
.flexContainer {
  display: flex;
  flex-wrap: wrap;
  margin: 50px auto;
  padding: 0px 30px;
  justify-content: center;
}

.flexItem {
  width: 300px;
  max-width: 80%;
  height: auto;
  margin: 30px;
}

.image {
  width: 80%;
  max-width: 240px;
  margin: 20px;
}

.previewImgStyle {
  max-width: 150px;
  max-height: 100px;
  margin: 20px;
  vertical-align: middle;
}

.note {
  color: gray;
}

.infoStyle {
  color: deepskyblue;
  margin-right: 5px;
}

.warningStyle {
  color: orange;
  margin-right: 5px;
}

.warningBox {
  margin: 20px;
}
