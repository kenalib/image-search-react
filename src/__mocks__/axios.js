import default_response from './default_response.json'

export default {
  get: jest.fn((url) => {
    return Promise.resolve({ data: default_response })
  }),
  post: jest.fn((url) => {
    return Promise.reject(new Error("Network Error"))
  }),
};
