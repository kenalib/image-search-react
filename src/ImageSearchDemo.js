import React, { Component } from 'react';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from '@material-ui/core/FormControl';
import axios from 'axios';
import constants from './Constants';

const flexContainer = {
  display: "flex",
  flexWrap: "wrap",
  margin: "50px auto",
  padding: "0px 50px",
}
const flexItem = {
  width: "300px",
  height: "auto",
  margin: "20px",
}
const imageStyle = {
  maxWidth: "240px",
}
const formControl = {
  margin: "30px",
  minWidth: "200px",
};

function FlexBox(props) {
  return (
    <div style={flexItem}>
      <Paper>
        <img src={constants.oss_url + props.item.picName} alt={""} style={imageStyle}/>
        <Typography gutterBottom variant="headline" component="h2">
          {props.item.picName}
        </Typography>
        <Typography component="p">
          {props.item.sortExprValues}
        </Typography>
      </Paper>
    </div>
  );
}

class ImageSearchDemo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      items: [],
      selectedFile: null,
      cat_id: "",
    }
  }

  componentDidMount() {
    fetch(constants.api_url)
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            items: result.SearchItemResponse.auctions
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          console.log("ERR connect api_url: " + error.message);

          this.setState({
            isLoaded: true,
            items: constants.default_result.SearchItemResponse.auctions
          });
        }
      )
  }

  fileChangedHandler = (event) => {
    this.setState({selectedFile: event.target.files[0]});
  }

  catChangeHandler = (event) => {
    this.setState({cat_id: event.target.value}, () => {
      this.uploadHandler();
    });
  }

  uploadHandler = () => {
    if (this.state.selectedFile === null) {
      console.log("ERR: image not set");
      return;
    }

    const formData = new FormData();
    formData.append('file_name', this.state.selectedFile, this.state.selectedFile.name);
    formData.append('cat_id', this.state.cat_id);

    axios.post(constants.api_url, formData).then(response => {
      this.setState((prevState, props) => {
        return { items: response.data.SearchItemResponse.auctions };
      });
    }).catch(err => {
      console.log("ERR: Service Suspended");
      console.log(err);
    });
  }

  render() {
    const { error, isLoaded, items } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    } else {
      return (
        <div>
          <Button variant="raised" label='My Label'>
            <input type="file" onChange={this.fileChangedHandler}/>
          </Button>
          <Button  variant="raised"
            onClick={this.uploadHandler}>Upload!</Button>

          <FormControl style={formControl}>
            <InputLabel htmlFor="category-simple">Category</InputLabel>
            <Select value={this.state.cat_id} onChange={this.catChangeHandler}
              inputProps={{name: 'cat_id', id: 'category-simple',}}>
              <MenuItem value="">Not Specified</MenuItem>
              {constants.allCategory.map(item => (
                <MenuItem value={item.id} key={item.id}>{item.id}: {item.name}</MenuItem>
              ))}
            </Select>
          </FormControl>

          <div style={flexContainer}>
            {items.map(item => (
              <FlexBox item={item} key={item.picName}/>
            ))}
          </div>
        </div>
      );
    }
  }
}

export default ImageSearchDemo;
