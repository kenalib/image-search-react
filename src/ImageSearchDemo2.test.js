import React from 'react';
import { shallow } from 'enzyme';
import ImageSearchDemo from './ImageSearchDemo';
import axios from 'axios';

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

it('BookList snapshot test', async () => {
  const wrapper = shallow(<ImageSearchDemo />);
  const instance = wrapper.instance();

  await flushPromises();
  wrapper.update();
  expect(wrapper).toMatchSnapshot();

  const fileContents = 'dummy';
  const file = new Blob([fileContents], {type : 'text/plain'});
  wrapper.find('input[type="file"]').simulate('change', {target: {files: [file]}});

  await flushPromises();
  await wrapper.update();
  expect(wrapper).toMatchSnapshot();

  expect(axios.post).toHaveBeenCalled();
  expect(wrapper).not.toContainReact(<div>Loading...</div>);
  expect(wrapper.text()).toMatch(/Network Error/);
});
