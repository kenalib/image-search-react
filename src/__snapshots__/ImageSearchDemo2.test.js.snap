// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList snapshot test 1`] = `
<div>
  <img
    alt=""
    className="previewImgStyle"
    src=""
  />
  <br />
  <WithStyles(Button)
    label="Upload Image"
    variant="raised"
  >
    <input
      onChange={[Function]}
      type="file"
    />
  </WithStyles(Button)>
  <WithStyles(FormControl)
    style={
      Object {
        "margin": "30px",
        "minWidth": "200px",
      }
    }
  >
    <WithStyles(InputLabel)
      htmlFor="category-simple"
    >
      Category
    </WithStyles(InputLabel)>
    <WithStyles(Select)
      inputProps={
        Object {
          "id": "category-simple",
          "name": "cat_id",
        }
      }
      onChange={[Function]}
      value=""
    >
      <WithStyles(MenuItem)
        key="0"
        value=""
      >
        Not Specified
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="0"
        value="0"
      >
        0
        : 
        Tops
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="1"
        value="1"
      >
        1
        : 
        Dress
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="2"
        value="2"
      >
        2
        : 
        Bottoms
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="3"
        value="3"
      >
        3
        : 
        Bag
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="4"
        value="4"
      >
        4
        : 
        Shoes
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="5"
        value="5"
      >
        5
        : 
        Accessories
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="6"
        value="6"
      >
        6
        : 
        Snack
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="7"
        value="7"
      >
        7
        : 
        Makeup
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="8"
        value="8"
      >
        8
        : 
        Bottle
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="9"
        value="9"
      >
        9
        : 
        Furniture
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="20"
        value="20"
      >
        20
        : 
        Toy
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="21"
        value="21"
      >
        21
        : 
        Underwear
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="22"
        value="22"
      >
        22
        : 
        Digital device
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="88888888"
        value="88888888"
      >
        88888888
        : 
        Other
      </WithStyles(MenuItem)>
    </WithStyles(Select)>
  </WithStyles(FormControl)>
  <div
    className="note"
  >
    note: demo data is in
    <br />
    1: Dress, 3: Bag, 4: Shoes
  </div>
  <div
    className="flexContainer"
  >
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1007",
          "picName": "bag02.jpg",
          "sortExprValues": "5.37633353624177e+24;0",
        }
      }
      key="bag02.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1010",
          "picName": "bag05.jpg",
          "sortExprValues": "3.37476205825806;225",
        }
      }
      key="bag05.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1009",
          "picName": "bag04.jpg",
          "sortExprValues": "3.29809403419495;219",
        }
      }
      key="bag04.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1006",
          "picName": "bag01.jpg",
          "sortExprValues": "2.99806690216064;242",
        }
      }
      key="bag01.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1008",
          "picName": "bag03.jpg",
          "sortExprValues": "2.84266543388367;233",
        }
      }
      key="bag03.jpg"
    />
  </div>
</div>
`;

exports[`BookList snapshot test 2`] = `
<div>
  <img
    alt=""
    className="previewImgStyle"
    src="data:text/plain;charset=undefined,dummy"
  />
  <br />
  <WithStyles(Button)
    label="Upload Image"
    variant="raised"
  >
    <input
      onChange={[Function]}
      type="file"
    />
  </WithStyles(Button)>
  <WithStyles(FormControl)
    style={
      Object {
        "margin": "30px",
        "minWidth": "200px",
      }
    }
  >
    <WithStyles(InputLabel)
      htmlFor="category-simple"
    >
      Category
    </WithStyles(InputLabel)>
    <WithStyles(Select)
      inputProps={
        Object {
          "id": "category-simple",
          "name": "cat_id",
        }
      }
      onChange={[Function]}
      value=""
    >
      <WithStyles(MenuItem)
        key="0"
        value=""
      >
        Not Specified
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="0"
        value="0"
      >
        0
        : 
        Tops
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="1"
        value="1"
      >
        1
        : 
        Dress
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="2"
        value="2"
      >
        2
        : 
        Bottoms
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="3"
        value="3"
      >
        3
        : 
        Bag
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="4"
        value="4"
      >
        4
        : 
        Shoes
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="5"
        value="5"
      >
        5
        : 
        Accessories
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="6"
        value="6"
      >
        6
        : 
        Snack
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="7"
        value="7"
      >
        7
        : 
        Makeup
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="8"
        value="8"
      >
        8
        : 
        Bottle
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="9"
        value="9"
      >
        9
        : 
        Furniture
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="20"
        value="20"
      >
        20
        : 
        Toy
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="21"
        value="21"
      >
        21
        : 
        Underwear
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="22"
        value="22"
      >
        22
        : 
        Digital device
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="88888888"
        value="88888888"
      >
        88888888
        : 
        Other
      </WithStyles(MenuItem)>
    </WithStyles(Select)>
  </WithStyles(FormControl)>
  <div
    className="note"
  >
    note: demo data is in
    <br />
    1: Dress, 3: Bag, 4: Shoes
  </div>
  <pre>
    Error: 
    Network Error
  </pre>
  <div
    className="flexContainer"
  >
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1007",
          "picName": "bag02.jpg",
          "sortExprValues": "5.37633353624177e+24;0",
        }
      }
      key="bag02.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1010",
          "picName": "bag05.jpg",
          "sortExprValues": "3.37476205825806;225",
        }
      }
      key="bag05.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1009",
          "picName": "bag04.jpg",
          "sortExprValues": "3.29809403419495;219",
        }
      }
      key="bag04.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1006",
          "picName": "bag01.jpg",
          "sortExprValues": "2.99806690216064;242",
        }
      }
      key="bag01.jpg"
    />
    <FlexBox
      item={
        Object {
          "catId": "3",
          "custContent": "k1:v1,k2:v2,k3:v3",
          "itemId": "1008",
          "picName": "bag03.jpg",
          "sortExprValues": "2.84266543388367;233",
        }
      }
      key="bag03.jpg"
    />
  </div>
</div>
`;
